orchagent/aclorch.cpp

```diff
@@ -1754,30 +1856,50 @@ bool AclOrch::processPorts(string portsList, std::function<void (sai_object_id_t
         Port port;
         if (!gPortsOrch->getPort(alias, port))
         {
-            SWSS_LOG_ERROR("Failed to process port. Port %s doesn't exist", alias.c_str());
-            return false;
+            SWSS_LOG_WARN("Port %s not configured yet, add it to ACL table %s pending list", alias.c_str(), aclTable.description.c_str());
+            aclTable.pendingPortSet.emplace(alias);
+            continue;
         }
 
-        switch (port.m_type)
+        port_id = getValidPortId(alias, port);
+        if (port_id != SAI_NULL_OBJECT_ID)
         {
-        case Port::PHY:
-            if (port.m_lag_member_id != SAI_NULL_OBJECT_ID)
-            {
-                SWSS_LOG_ERROR("Failed to process port. Bind table to LAG member %s is not allowed", alias.c_str());
-                return false;
-            }
-            inserter(port.m_port_id);
-            break;
-        case Port::LAG:
-            inserter(port.m_lag_id);
-            break;
-        case Port::VLAN:
-            inserter(port.m_vlan_info.vlan_oid);
-            break;
-        default:
-          SWSS_LOG_ERROR("Failed to process port. Incorrect port %s type %d", alias.c_str(), port.m_type);
-          return false;
-      }
+            inserter(port_id);
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool AclOrch::processPendingPort(AclTable &aclTable, string portAlias, std::function<void (sai_object_id_t)> inserter)
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_INFO("Processing ACL table port %s", portAlias.c_str());
+
+    sai_object_id_t port_id;
+
+    Port port;
+    if (!gPortsOrch->getPort(portAlias, port))
+    {
+        SWSS_LOG_WARN("Port %s not configured yet, add it to ACL table %s pending list", portAlias.c_str(), aclTable.description.c_str());
+        aclTable.pendingPortSet.insert(portAlias);
+        return true;
+    }
+
+    port_id = getValidPortId(portAlias, port);
+    if (port_id != SAI_NULL_OBJECT_ID)
+    {
+        inserter(port_id);
+        aclTable.bind(port_id);
```

Comment:
>aclTable.bind(port_id); [](start = 8, length = 23)

move this out of processPendingPort() so that you can reuse this function.

Reply:
I have clarified with Qi before, my view is that these two functions do share some same code, but since processPorts is for "normal" flow and for processPendingPort is dedicated to receiving the notification and handle the pending port, to keep the code more straightforward, I feel like to have these two functions separated though will have some slight redundant code.